pandas/core/frame.py

```diff
@@ -953,16 +953,61 @@ def __len__(self):
 
     def dot(self, other):
         """
-        Matrix multiplication with DataFrame or Series objects.  Can also be
-        called using `self @ other` in Python >= 3.5.
+        Compute the matrix mutiplication between the DataFrame and other.
+
+        This method computes the matrix product between the DataFrame and the
+        values of an other Series, DataFrame or a numpy array.
+
+        It can also be called using `self @ other` in Python >= 3.5.
 
         Parameters
         ----------
-        other : DataFrame or Series
+        other : Series, DataFrame or array-like
+            The other object to compute the matrix product with.
 
         Returns
         -------
-        dot_product : DataFrame or Series
+        Series or DataFrame
+            Return the Series of the matrix product between the Dataframe and
+            other if other is a Series, the Dataframe of the matrix product of
+            each columns of the DataFrame/np.array and each columns of other
+            if other is a DataFrame or a numpy.ndarray.
```

Comment:
Can you rephrase this? Can't really understand it, we probably need to have more than one sentence.